playwright/src/main/java/com/microsoft/playwright/impl/VideoImpl.java

```diff
@@ -16,27 +16,67 @@
 
 package com.microsoft.playwright.impl;
 
+import com.microsoft.playwright.PlaywrightException;
 import com.microsoft.playwright.Video;
 
 import java.nio.file.Path;
+import java.nio.file.Paths;
 
-class VideoImpl implements Video {
+import static java.util.Arrays.asList;
+
+class VideoImpl extends LoggingSupport implements Video {
   private final PageImpl page;
-  private Path fullPath;
+  private final WaitableResult<ArtifactImpl> waitableArtifact = new WaitableResult<>();
+  private final boolean isRemote;
 
   VideoImpl(PageImpl page) {
     this.page = page;
+    BrowserImpl browser = page.context().browser();
+    isRemote = browser != null && browser.isRemote;
```

Comment:
if (page.isClosed())
  waitableArtifact.complete(null);

Reply:
This is unnecessary, waitForArtifact already does it.